osv/ecosystems/_ecosystems.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Ecosystem helpers."""
+
+from .helper_base import Ecosystem, OrderingUnsupportedEcosystem
+from .alpine import Alpine
+from .debian import Debian
+from .maven import Maven
+from .nuget import NuGet
+from .packagist import Packagist
+from .pub import Pub
+from .pypi import PyPI
+from .rubygems import RubyGems
+from .semver import Crates, Go, Hex, NPM
+
+_ecosystems = {
+    'crates.io': Crates(),
+    'Go': Go(),
+    'Hex': Hex(),
+    'Maven': Maven(),
+    'npm': NPM(),
+    'NuGet': NuGet(),
+    'Packagist': Packagist(),
+    'Pub': Pub(),
+    'PyPI': PyPI(),
+    'RubyGems': RubyGems(),
+    # Ecosystems missing implementations:
+    'Android': OrderingUnsupportedEcosystem(),
+    'GitHub Actions': OrderingUnsupportedEcosystem(),
+    'Linux': OrderingUnsupportedEcosystem(),
+    'OSS-Fuzz': OrderingUnsupportedEcosystem(),
+    # Alpine and Debian require a release version for enumeration, which is
+    # handled separately in get().
+    'Alpine': OrderingUnsupportedEcosystem(),
+    'Debian': OrderingUnsupportedEcosystem(),
+}
+
+SEMVER_ECOSYSTEMS = {
+    'crates.io',
+    'Go',
+    'npm',
+    'Hex',
+}
+
+
+def get(name: str) -> Ecosystem:
```

Comment:
Suggestion: render this file unnecessary by moving `get()` and `normalize()` into `helper_base.py` and the rest of it into `__init__py` (which would address some of my discoverability/readability concerns)

Reply:
I don't think I can move `get()` into `helper_base` without creating circular imports (`get` needs the `ecosystems_` dictionary, which needs to import the ecosystems implementations, which needs to import the base class)

I could move everything from `_ecosystems.py` into `__init__.py`, although I personally feel that makes it less discoverable (to me, it's not always obvious that `__init__.py` would contain actual code).

Reply:
Ah, fair enough. Never mind then.